### PR TITLE
WT-15213 Don't free the previous address cookie

### PR DESCRIPTION
**Pull request by mm-ng**

### CODE REVIEW

**Comment by quchenhao** [comment]
@mm-ng Can we create test for this?

File: src/reconcile/rec_write.c [hunk 1]
```diff
@@ -2671,8 +2671,6 @@ __rec_copy_prev_addr(WT_SESSION_IMPL *session, WTI_RECONCILE *r)
               mod->mod_replace.block_cookie_size, &multi->addr.block_cookie));
             multi->addr.block_cookie_size = mod->mod_replace.block_cookie_size;
             multi->addr.type = mod->mod_replace.type;
-            __wt_free(session, mod->mod_replace.block_cookie);
```
Comment (kommiharibabu) [inline src/reconcile/rec_write.c:2674]: We already copied this address into the `multi->addr.block_cookie` a couple of lines above. May be we need to use the above pointer somewhere we are accessing?

**Comment by mm-ng** [comment]
> @mm-ng Can we create test for this?

The test/format config reproduces this consistently, I can add this into the PR if you want?

Or did you want a more deterministic test that attempts to generate multiple empty deltas in a row?

**Comment by quchenhao** [comment]
> > @mm-ng Can we create test for this?
> 
> The test/format config reproduces this consistently, I can add this into the PR if you want?
> 
> Or did you want a more deterministic test that attempts to generate multiple empty deltas in a row?

Can we first try write a python test? I prefer a deterministic test.

File: src/reconcile/rec_write.c (hunk 1 above)
Comment (kommiharibabu) [reply src/reconcile/rec_write.c:2674]: Chenhao explained the need for saving this address to handle the reconciliation failure and then building an empty delta.

**Comment by mm-ng** [comment]
@quchenhao @kommiharibabu Are we ok to merge in this fix? I spoke to Sid and he doesn't want us to merge in the failing CONFIG because there are currently other disagg issues present that can create more CI-blockers. However, I can file a ticket and add it to his testing project to add the CONFIG into the failure configs folder when things are stable.

Edit: New ticket [here](https://jira.mongodb.org/browse/WT-15414).

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by quchenhao — APPROVED** [review]
lgtm
